modules/plugins/visuals/fidget/fidget.nix

```diff
@@ -0,0 +1,374 @@
+                "dots"
+                "dots_negative"
+                "dots_snake"
+                "dots_footsteps"
+                "dots_hop"
+                "line"
+                "pipe"
+                "dots_ellipsis"
+                "dots_scrolling"
+                "star"
+                "flip"
+                "hamburger"
+                "grow_vertical"
+                "grow_horizontal"
+                "noise"
+                "dots_bounce"
+                "triangle"
+                "arc"
+                "circle"
+                "square_corners"
+                "circle_quarters"
+                "circle_halves"
+                "dots_toggle"
+                "box_toggle"
+                "arrow"
+                "zip"
+                "bouncing_bar"
+                "bouncing_ball"
+                "clock"
+                "earth"
+                "moon"
+                "dots_pulse"
+                "meter"
+              ];
+              default = "dots";
+            };
+            period = mkOption {
+              description = "Period of the pattern";
+              type = int;
+              default = 1;
+            };
+          };
+          progress_style = mkOption {
+            description = "Highlight group for in-progress LSP tasks";
+            type = str;
+            default = "WarningMsg";
+          };
+          group_style = mkOption {
+            description = "Highlight group for group name (LSP server name)";
+            type = str;
+            default = "Title";
+          };
+          icon_style = mkOption {
+            description = "Highlight group for group icons";
+            type = str;
+            default = "Question";
+          };
+          priority = mkOption {
+            description = "Priority of the progress notification";
+            type = int;
+            default = 30;
+          };
+          skip_history = mkOption {
+            description = "Skip adding messages to history";
+            type = bool;
+            default = true;
+          };
+          format_message = mkOption {
+            description = "How to format a progress message";
+            type = luaInline;
+            default = mkLuaInline ''
+              require("fidget.progress.display").default_format_message
+            '';
+          };
+          format_annote = mkOption {
+            description = "How to format a progress annotation";
+            type = luaInline;
+            default = mkLuaInline ''
+              function(msg) return msg.title end
+            '';
+          };
+          format_group_name = mkOption {
+            description = "How to format a progress notification group's name";
+            type = luaInline;
+            default = mkLuaInline ''
+              function(group) return tostring(group) end
+            '';
+          };
+          overrides = mkOption {
+            description = "Override options from the default notification config";
```

Comment:
Can we make notification a submodule type and reuse it here? Makes it less cursed